### ISSUE TITLE
ASSERT failed: wc::linear_search

### ISSUE THREAD

**Issue by Makishima** [WarpzoneStudios/Hammerting#94]
0.2.1.5
https://1drv.ms/u/s!Asn3lEt-4tOhjp5l8UuL4mshY65TAw?e=o8QXRw

```
Hammerting version 0.2.1.5
Save version: version 0.2.1.5|C:\buildbot\build_tdtd-patch-windows-dev\warpcore\graph\public\graph/graph_util.h:355] ASSERT failed: wc::linear_search( ordered_allowed_edge_types, ordered_allowed_edge_types + edge_type_count, edges[pos->_edge]._type ) != nullptr [Couldn't find allowed edge type in current graph position edge

Hammerting version 0.2.1.5
Save version: version 0.2.1.5

Callstack:
0: graph_find_edge_for_graph_pos
graph\public\graph\graph_util.h:355:96
1: movement_ensure_pos_on_edge
tdtd\gamestate\movement_util.cpp:215:0
2: bt_fall_update
tdtd\gamestate\behavior_trees\node_fall.cpp:200:0
3: ai_behavior_system_update
tdtd\gamestate\systems\ai_behavior_system.cpp:331:0
4: entity_entity_manager_update
entity\entity_manager.cpp:183:0
5: gamestate::update
tdtd\gamestate\gamestate.cpp:634:0
6: main_main_h::update
tdtd\main\main.cpp:561:0
7: game_func
tdtd\main\main.cpp:935:0
8: thread_entry
wc_sys\thread.cpp:104:0
9: configthreadlocale
10: BaseThreadInitThunk
11: RtlUserThreadStart
]
```

**Comment by smokingwreckage** [comment]
---------------------------
tdtd: The game has crashed.
---------------------------
Error: Couldn't find allowed edge type in current graph position edge

Hammerting version 0.2.1.6
Save version: version 0.2.1.4,version 0.2.1.6

Callstack:
0: graph_find_edge_for_graph_pos
graph\public\graph\graph_util.h:355:96
1: movement_ensure_pos_on_edge
tdtd\gamestate\movement_util.cpp:215:0
2: bt_fall_update
tdtd\gamestate\behavior_trees\node_fall.cpp:200:0
3: ai_behavior_system_update
tdtd\gamestate\systems\ai_behavior_system.cpp:331:0
4: entity_entity_manager_update
entity\entity_manager.cpp:183:0
5: gamestate::update
tdtd\gamestate\gamestate.cpp:634:0
6: main_main_h::update
tdtd\main\main.cpp:561:0
7: game_func
tdtd\main\main.cpp:935:0
8: thread_entry
wc_sys\thread.cpp:104:0
9: configthreadlocale
10: BaseThreadInitThunk
11: RtlUserThreadStart


Assertion failed: wc::linear_search( ordered_allowed_edge_types, ordered_allowed_edge_types + edge_type_count, edges[pos->_edge]._type ) != nullptr

File: C:\buildbot\build_tdtd-patch-windows-dev\warpcore\graph\public\graph/graph_util.h
Line: 355

We would love to see this error message, so we can have a chance to fix the problem. 

If you want to help us with this, please press CTRL+C to copy this text, then send it to bugs@warpzonestudios.com or join us on Discord and show us there: https://discord.gg/pbT9zhw


---------------------------
OK   
---------------------------

**Comment by Srekel** [comment]
Fixed in 0.2.1.8
